### PR TITLE
refactors MetaVote and creates unit tests for it

### PR DESCRIPTION
**Pull request by maqi**
This PR contains the work of refactor the previous MetaVote to separate concerns.
Also, unit tests for the new struct MetaVoteValues, and unit tests for the refactored MetaVotes are created to improve the test coverage.

### CODE REVIEW

File: src/meta_voting/meta_vote_values.rs [hunk 1]
```diff
@@ -0,0 +1,910 @@
...
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use super::{bool_set::BoolSet, meta_vote_counts::MetaVoteCounts};
+use std::fmt::{self, Debug, Formatter};
+use std::num::NonZeroUsize;
+
+#[derive(Clone, Copy, Default, PartialEq, Eq, Serialize, Deserialize)]
+pub(crate) struct Estimates(BoolSet);
```
Comment (pierrechevalier83) [inline src/meta_voting/meta_vote_values.rs]: It appears all this code is duplicated between meta_vote.rs and meta_vote_values.rs

Comment (pierrechevalier83) [reply src/meta_voting/meta_vote_values.rs]: Looking at the next commit, it's a git mishap.
I would say: either squash both commits and mention refactoring and testing in the commit message or make it two commits, but then you need to move the tests to the second commit and put the new MetaVoteValues and friends in the right place in the first commit

Comment (maqi) [reply src/meta_voting/meta_vote_values.rs]: I will still make it as two commits, but move some of the refactoring part from the second commit to the first commit.

File: src/meta_voting/meta_vote_counts.rs [hunk 2]
```diff
@@ -53,7 +39,7 @@ impl MetaVoteCounts {
                     .filter(|vote| vote.round_and_step() == parent.round_and_step())
                     .last()
             })
-            .chain(if is_voter { Some(parent) } else { None })
+            .chain(Some(parent))
```
[the remark this answers is not in the document]
Comment (maqi) [reply src/meta_voting/meta_vote_counts.rs]: I will update to use `iter::once` once got the comments from Pierre.

**Pull request merged**